File: Python solutions/861. Score After Flipping Matrix.py
class Solution:
    def binary_to_decimal(self, s: str) -> int:
        decimal = 0
        for digit in s:
            decimal = (decimal * 2) + int(digit)
        return decimal

    def matrixScore(self, grid: List[List[int]]) -> int:
        n = len(grid)
        for i in range(n):
            if grid[i][0] != 1:
                m = len(grid[i])
                for j in range(m):
                    if grid[i][j] == 0:
                        grid[i][j] = 1
                    else:
                        grid[i][j] = 0
        
        m = len(grid[0]) 
        for i in range(m):
            cnt_zero = cnt_one = 0
            for j in range(n):
                if grid[j][i] == 0:
                    cnt_zero += 1
                else:
                    cnt_one += 1  

            if cnt_zero > cnt_one:
                for j in range(n):
                    if grid[j][i] == 0:
                        grid[j][i] = 1
                    else:
                        grid[j][i] = 0       


        ans = 0
        for i in range(n):
            m = len(grid[i])
            s = ""
            for j in range(m):
                s += str(grid[i][j])

            ans += self.binary_to_decimal(s)
        return ans

File: C++ solutions/861. Score After Flipping Matrix.cpp
class Solution {
public:
    int binary_to_decimal(string s) {
        int decimal = 0;
        for (char digit : s) {
            decimal = (decimal * 2) + (digit - '0');
        }
        return decimal;
    }

    int matrixScore(vector<vector<int>>& grid) {
        int n = grid.size();
        for (int i = 0; i < n; ++i) {
            if (grid[i][0] != 1) {
                int m = grid[i].size();
                for (int j = 0; j < m; ++j) {
                    if (grid[i][j] == 0) {
                        grid[i][j] = 1;
                    } else {
                        grid[i][j] = 0;
                    }
                }
            }
        }

        int m = grid[0].size();
        for (int i = 0; i < m; ++i) {
            int cnt_zero = 0, cnt_one = 0;
            for (int j = 0; j < n; ++j) {
                if (grid[j][i] == 0) {
                    cnt_zero++;
                } else {
                    cnt_one++;
                }
            }

            if (cnt_zero > cnt_one) {
                for (int j = 0; j < n; ++j) {
                    if (grid[j][i] == 0) {
                        grid[j][i] = 1;
                    } else {
                        grid[j][i] = 0;
                    }
                }
            }
        }

        int ans = 0;
        for (int i = 0; i < n; ++i) {
            string s = "";
            for (int j = 0; j < m; ++j) {
                s += to_string(grid[i][j]);
            }
            ans += binary_to_decimal(s);
        }
        return ans;
    }
};